Simulator/Core/Core.cpp

```diff
@@ -190,6 +190,9 @@ int Core::runSimulation(string executableName, string cmdLineArguments)
       }
    }
 
+   // Helper function for recorder to register spike history variables for all neurons.
+   simulator.getModel().getLayout().getVertices().registerHistoryVariables();
+
```

Comment:
Isn't this already called from either the vertex constructor or read params or something? Otherwise, how is the registration happening?

Reply:
Before this change, `registerSpikeHistoryVariable` (now `registerHistoryVariables`) was a member function of `AllSpikingNeurons` and was called within `AllSpikingNeurons::setupVertices()`.

The main reason for moving this function to the `AllVertices` base class is to accommodate deserialization. Previously, our flow was `setup -> register -> deserialize -> simulate`, and recording variables during the setup stage meant missing out on the deserialized vertices.

With this change, the new flow is `setup -> deserialize -> register -> simulate`, ensuring that all relevant variables, including those from deserialized vertices, are properly recorded.

Reply:
This is good; I will add it to issues like #131